students/BrianNisonger/Lesson5/assignment/test_database.py

```diff
@@ -0,0 +1,177 @@
+import os
+import pytest
+
+from database import import_data
+from database import show_available_products
+from database import show_rentals
+from database import calculate_availability
+from database import make_product_dict
+from database import return_user_ids
+from database import make_customer_dict
+
+from pymongo import MongoClient
+
+
+class MongoDBConnection(object):
+    """MongoDB Connection"""
+
+    def __init__(self, host='127.0.0.1', port=27017):
+        """ be sure to use the ip address not name for local windows"""
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.close()
+
+
+@pytest.fixture
+def setup_db(request):
+    mongo = MongoDBConnection()
+    with mongo:
+        db = mongo.connection.furniture
+        db.product_data.drop()
+        db.customer_data.drop()
+        db.rental_data.drop()
+        import_data("dat", "product_data.csv", "customer_data.csv",
+                    "rental_data.csv")
+
+        def cleanup():
+            db.product_data.drop()
+            db.customer_data.drop()
+            db.rental_data.drop()
+
+        request.addfinalizer(cleanup)
+
+
+@pytest.fixture
+def _show_available_products():
+    product_dict = {
+        'prd1001': {
+            'description': '60 inch TV stand',
+            'product_type': 'livingroom',
+            'quantity_available': 2
+        },
+        'prd1003': {
+            'description': 'Hobart Stand Mixer',
+            'product_type': 'kitchen',
+            'quantity_available': 10
+        }
+    }
+    return product_dict
```

Comment:
It is a little confusing calling this show_available_products. Why is it a function?